pkg/build/generator/generator.go

```diff
@@ -353,6 +363,13 @@ func (g *BuildGenerator) Clone(ctx kapi.Context, request *buildapi.BuildRequest)
 	newBuild := generateBuildFromBuild(build, buildConfig)
 	glog.V(4).Infof("Build %s/%s has been generated from Build %s/%s", newBuild.Namespace, newBuild.ObjectMeta.Name, build.Namespace, build.ObjectMeta.Name)
 
+	// Add the original Build's OwnerReferences to the new Builds. This will
+	// make sure that the new Build is deleted when the original BuildConfig
+	// is deleted.
+	if len(build.OwnerReferences) > 0 {
```

Comment:
I think this would be better as a one-liner (I don't think the if statement is necessary) in generateBuildFromBuild, where newBuild is being instantiated.